src/AbstractOperations/function_fields.jl

```diff
@@ -0,0 +1,30 @@
+struct FunctionField{X, Y, Z, C, F, G} <: AbstractLocatedField{X, Y, Z, F, G}
+     func :: F
+     grid :: G
+    clock :: C
+
+    function FunctionField{X, Y, Z}(func, grid; clock=nothing) where {X, Y, Z}
+        return new{X, Y, Z, typeof(clock), typeof(func), typeof(grid)}(func, grid, clock)
+    end
+end
```

Comment:
> I'm guessing this is a Field that is evaluated on-the-fly via a function so that it doesn't use up physical memory for a 3D array?

Yes. It's a `Field` which returns the result of a function evaluated at a particular location (and time, if it knows `clock`) when indexed at `i, j, k`.